media/js/privacy/firefox-privacy.js

```diff
@@ -0,0 +1,288 @@
+            property: 'data-parent-container'
+        });
+        var isExpanded = dataAttributeHandler({
+            elem: targetElem,
+            property: 'data-is-expanded'
+        });
+
+        var targetAction = isExpanded === 'true' ? 'add' : 'remove';
+        var targetSection = document.getElementById(targetSectionId);
+        var subSections = targetSection.querySelectorAll('section');
+
+        for (var i = 0, l = subSections.length; i < l; i++) {
+            var divElem = subSections[i].querySelector('div');
+            var heading = subSections[i].querySelector('h3');
+
+            divElem.classList[targetAction]('hidden');
+            divElem.setAttribute('aria-hidden', isExpanded || 'false');
+            heading.classList[targetAction]('collapsed');
+        }
+
+        if (isExpanded === 'true') {
+            dataAttributeHandler({
+                elem: targetElem,
+                property: 'data-is-expanded',
+                value: 'false'
+            });
+
+            targetElem.textContent = tabpanelOpenText;
+        } else {
+            dataAttributeHandler({
+                elem: targetElem,
+                property: 'data-is-expanded',
+                value: 'true'
+            });
+
+            targetElem.textContent = tabpanelCloseText;
+        }
+    }
+
+    /**
+     * Listens for click events on items inside the main content area.
+     * It then calls the appropriate function based on the event target.
+     * This avoids adding multiple event handlers on individual elements.
+     */
+    mainContent.addEventListener('click', function(event) {
+        // hanle clicks on an individual topic
+        if (event.target.tagName === 'H3') {
+            var tabContent = event.target.parentElement.nextElementSibling;
+            if (tabContent.classList.contains('hidden')) {
+                tabContent.classList.remove('hidden');
+                tabContent.setAttribute('aria-hidden', false);
+                event.target.classList.remove('collapsed');
+            } else {
+                tabContent.classList.add('hidden');
+                tabContent.setAttribute('aria-hidden', true);
+                event.target.classList.add('collapsed');
+            }
+        }
+
+        // handle clicks on the master toggle buttons
+        if (event.target.classList.contains('toggle')) {
+            toggleMainSectionTopics(event.target);
+        }
+
+        // handle clicks on the data choices "Choose" button
+        if (event.target.id === 'choose') {
+            // if the uitour did not load, just return
+            if (Mozilla.UITour === undefined) {
+                return;
+            }
+
+            Mozilla.UITour.openPreferences('privacy-reports');
+        }
+    });
+
+    /* add a class to indicate that js is enabled. This will trigger
+    the appropriate styling to be applied */
+    mainContent.classList.add('interactive-widget');
+
+    initPage();
```

Comment:
All this vanilla JS is great 👍 , but I'd suggest using a feature detect just to stop older browsers throwing a bunch of errors?